### PR TITLE
Updated for multi-module issues and branch-sync issues

### PR DESCRIPTION
**Pull request by saleem-latif**
__Description:__
Here is a list of changes included in this PR.

1. Fixed handling for multi-module projects in languages that support modules such as java. These were being considered as separate repo by the plugin.
2. Added fixes for sync prompt at the start, it was incorrectly prompting user to sync the repo if user creates a new branch offline for an already synced repo.
3. Fixed invalid payload issue for branch syncing process, specifically `is_public` was missing and causing an error.
4. Improved messaging of repo in sync message.

**Testing Instructions:**
- Sync a new repo. Make sure it gets synced successfully.
- Disconnect a repo and then close and open the project without closing the IDE, user should be prompted to sync the repo again.
-  Disconnect the repo and sync using codesync menu.
- Change the branch of a repo and make sure new branch gets uploaded
- Close the project create another new branch and start the project again. make sure it does not ask to start the syncing again. And syncs the new branch in the background.

### CODE REVIEW

File: src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java [hunk 1]
```diff
@@ -177,9 +176,10 @@ public static void setupCodeSyncRepo(Project project, String repoPath, String re
 
                     if (shouldSyncRepo) {
                         syncRepo(repoPath, repoName, branchName, project, codeSyncProgressIndicator, false);
+                        reposBeingSynced.remove(repoPath);
                     }
                 }
-            } else if (!configFile.isRepoDisconnected(repoPath)) {
+            } else if (!isSyncingBranch) {
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java:182]: Should `!configFile.isRepoDisconnected(repoPath)` be replaced with `configFile.isRepoActive(repoPath)`?

File: src/test/java/TestUtils.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import com.intellij.openapi.application.PathManager
+import com.intellij.openapi.util.io.FileUtil
+
+object CodeSyncTestUtils {
+    fun getTestDataPath(): String {
+        return FileUtil.toCanonicalPath("${PathManager.getTempPath()}/")
```
Comment (Basit-qc) [inline src/test/java/TestUtils.kt:6]: How about keeping it inside `tests` directory as `tests/tests_data/` and also add it to .gitignore/.syncignore?

**Review by Basit-qc — APPROVED** [review]
Looks great 👍 added couple questions for my understanding.

File: src/test/java/TestUtils.kt (hunk 2 above)
Comment (saleem-latif) [reply src/test/java/TestUtils.kt:6]: I tried doing that, but I could not find a way to get the path to `tests/tests_data/` directory from the code. I will do more search later and change this if necessary.

Comment (Basit-qc) [reply src/test/java/TestUtils.kt:6]: @syedgulahmed has done this as https://github.com/codesyncapp/codesync-intellij/pull/78/files#diff-ab7386c3800fb9121247645db6fa87a597432991a80c364a60cc56fb01a45bcdR6 (PR is still open) Maybe that helps.

File: src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java (hunk 1 above)
Comment (saleem-latif) [reply src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java:182]: No, using `! isSyncingBranch` is a better option here.

**Pull request merged**